lib/Transform/XTenMinimizeLiveTensors.cpp

```diff
@@ -209,6 +198,24 @@ class XTenMinimizeLiveTensorsPass
     }
   }
 
+  /// Checks for illegal dead code by traversing operations and
+  /// verifying if they are dead and any of their operands match any OpInfo.
+  /// Dead code can be legal if it is not connected to operation collection
+  /// through 'OpInfo' data structure.
+  bool hasIllegalDeadCode() {
+    return llvm::any_of(currFn.getBody().getOps(), [&](Operation &op) {
+      auto hasUses = !op.use_empty();
```

Comment:
I don't understand the role of `hasUses` here. Is one of the tests below sensitive to this so I can understand by example?

Reply:
The operation, per se, won't be dead if it is used by another. All tests trigger this, since we are interested on situations that could cause the reordering to fail due to dead code.

Reply:
The correction looks good.